### PR TITLE
Add C++20 audit

### PR DESCRIPTION
**Pull request by andrewsavage1**
b/309847996

### CODE REVIEW

**Comment by andrewsavage1** [comment]
Doing an initial run on all platforms to see which pass vs fail

File: starboard/build/config/modular/BUILD.gn [hunk 1]
```diff
@@ -32,10 +32,13 @@ config("modular") {
...
   }
 
-  cflags_cc = [
-    "-nostdinc++",
-    "-std=c++17",
-  ]
+  cflags_cc = [ "-nostdinc++" ]
+
+  if (sb_enable_cpp20_audit) {
+    cflags_cc += [ "-std=c++17" ]
```
Comment (niranjanyardi) [inline starboard/build/config/modular/BUILD.gn:38]: 20 ?

**Review by niranjanyardi — APPROVED** [review]
lgtm depending on passing tests. gentle reminder : please run internal tests too.

File: starboard/nplb/compiler_compliance/cpp20_support.cc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// limitations under the License.
+
+#include <string>
+
+namespace starboard {
+namespace nplb {
+namespace compiler_compliance {
+namespace {
+
+// Test string ends with support
```
Comment (kaidokert) [inline starboard/nplb/compiler_compliance/cpp20_support.cc:22]: Would you mind adding a few language feature checks as well, in addition to library features? concepts, consteval and constinit would be good candidates to check for, it'll break really quick on old compilers:

https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#core-allowlist-20

**Review by kaidokert — APPROVED** [review]

**Comment by niranjanyardi** [comment]
folded into: https://github.com/youtube/cobalt/pull/2688

**Pull request closed without merging**